### PR TITLE
Playing with tests

### PR DESCRIPTION
**Pull request by Aruzal**
Played around with some testing.

### CODE REVIEW

**Comment by Aruzal** [comment]
Not sure why test in tests/api.js for / is failing?

**Comment by PapaPwn** [comment]
It's failing because /shizzlwazzle is returning status 200, not 404, which should supposedly be accounted for in the front end due to the comment:
```
/*
    anything that gets here and not handled
    by api error handler should get index.html.
    NB: frontend must handle actual 404s.
*/
app.use('*', function (_, res) {
    res.sendFile(path.join(__dirname, '../public', 'index.html'))
})
```
Not sure if the front end recognises that no data was returned from the request or if it works differently?

File: backend/tests/routes/api.js [hunk 1]
```diff
@@ -15,6 +15,12 @@ describe('API route testing', () => {
             .expect(200)
     )
 
+    it('/api/bad/path', () =>
```
Comment (DarkPurple141) [inline backend/tests/routes/api.js]: This one returns 404, but I think it was returning 200. I changed it.

File: backend/tests/routes/api.js [hunk 2]
```diff
@@ -15,6 +15,12 @@ describe('API route testing', () => {
...
+        supertest
+            .get('/api/bad/path')
+            .expect(404)
+    )
+
     it('Test fallback for bad path', () =>
```
Comment (DarkPurple141) [inline backend/tests/routes/api.js]: This should return 200.

Comment (DarkPurple141) [reply backend/tests/routes/api.js]: (And it does)

File: backend/tests/routes/course.js [hunk 3]
```diff
@@ -10,6 +11,48 @@ describe('Course route testing', () => {
...
+            .expect((req) => {
+                expect(req.body.courseID).to.be.a('number')
+                expect(req.body.courseCode).to.be.a('string')
+                expect(req.body.courseName).to.be.a('string')
+            })
+    )
+
+    describe('GET /api/course/1', () => {
+        let request
+
```
Comment (DarkPurple141) [inline backend/tests/routes/course.js]: See the way I've done this section. Apply to the above section.

**Pull request merged**